src/iuse.cpp

```diff
@@ -6458,20 +6562,809 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
     }
 };
 
+static std::string colorized_trap_name_at( const tripoint &point )
+{
+    const trap &trap = g->m.tr_at( point );
+    std::string name;
+    if( !trap.is_null() && trap.get_visibility() <= 1 ) {
+        name = colorize( trap.name(), trap.color ) + _( " on " );
+    };
+    return name;
+}
+
+static std::string colorized_field_description_at( const tripoint &point )
+{
+    static const std::unordered_set<field_id, std::hash<int>> covered_in_affix_ids = {
+        fd_blood, fd_bile, fd_gibs_flesh, fd_gibs_veggy, fd_web,
+        fd_slime, fd_acid, fd_sap, fd_sludge, fd_blood_veggy,
+        fd_blood_insect, fd_blood_invertebrate,  fd_gibs_insect,
+        fd_gibs_invertebrate, fd_rubble
+    };
+    static const std::unordered_set<field_id, std::hash<int>> on_affix_ids = {
+        fd_fire, fd_flame_burst
+    };
+    static const std::unordered_set<field_id, std::hash<int>> under_affix_ids = {
+        fd_gas_vent, fd_fire_vent, fd_fatigue
+    };
+    static const std::unordered_set<field_id, std::hash<int>> illuminated_by_affix_ids = {
+        fd_spotlight, fd_laser, fd_dazzling, fd_spotlight, fd_electricity
+    };
+    static const std::vector<std::pair<std::unordered_set<field_id, std::hash<int>>, std::string>>
+    affixes_vec = {
+        { covered_in_affix_ids, _( " covered in %s" ) },
```

Comment:
Affixes could be put to `field_types.json` (we can do it in follow-up PR).

Reply:
That will be better, i want to merge this asap, 30 days is too long for one feature